var/spack/repos/builtin/packages/kokkos/package.py

```diff
@@ -364,16 +368,10 @@ def cmake_args(self):
         if self.spec.satisfies("%oneapi") or self.spec.satisfies("%intel"):
             options.append(self.define("CMAKE_CXX_FLAGS", "-fp-model=precise"))
 
-        # Kokkos 4.2.00+ changed the default to Kokkos_ENABLE_IMPL_CUDA_MALLOC_ASYNC=on
-        # which breaks GPU-aware with Cray-MPICH
-        # See https://github.com/kokkos/kokkos/pull/6402
-        # TODO: disable this once Cray-MPICH is fixed
-        if (
-            self.spec.satisfies("@4.2.00:")
-            and "mpi" in self.spec
-            and self.spec["mpi"].name == "cray-mpich"
-        ):
-            options.append(self.define("Kokkos_ENABLE_IMPL_CUDA_MALLOC_ASYNC", False))
+        if self.spec.satisfies("+cuda"):
+            options.append(
+                self.define_from_variant("Kokkos_ENABLE_IMPL_CUDA_MALLOC_ASYNC", "alloc_async")
```

Comment:
Is it ok to define this before v4.2 ?

Reply:
This shouldn't matter, since if an unknown option is passed it'll not have an effect.

Reply:
True. My point is more that when you debug a failed build, and you see a bunch of warnings for unknown options, you don't know if that was intentional, or if it's just a badly written `package.py`.

Reply:
@cedricchevalier19 can you rewrite this? btw, you don't even have to use the `if` condition. Just change it to:
```python
options.append(self.define_from_variant("Kokkos_ENABLE_IMPL_CUDA_MALLOC_ASYNC", "alloc_async"))
```
Since `alloc_async` is a conditional variant, it'll produce an [empty string if the condition isn't met](https://github.com/spack/spack/blob/962262a1d3d2ca8db50fa82ec5684ba85b2a8c99/lib/spack/spack/build_systems/cmake.py#L522-L524).

Reply:
Sorry I've been busy. I will rewrite it today. Thanks for the suggestions @rbberger and @alalazo .